xline-client/src/lease_gen.rs

```diff
@@ -0,0 +1,84 @@
+use std::sync::atomic::{AtomicU64, Ordering};
+
+use clippy_utilities::Cast;
+
+/// Generator of unique lease id
+/// Note that this Lease Id generation method may cause collisions,
+/// the client should retry after informed by the server.
+#[derive(Debug, Default)]
+pub struct LeaseIdGenerator {
+    /// the current lease id
+    id: AtomicU64,
+}
+
+impl LeaseIdGenerator {
+    /// New `IdGenerator`
+    ///
+    /// # Panics
+    ///
+    /// panic if failed to generate random bytes
+    #[inline]
+    #[must_use]
+    pub fn new() -> Self {
+        let mut buf = [0u8; 8];
+        getrandom::getrandom(&mut buf).unwrap_or_else(|err| {
+            panic!("Failed to generate random bytes for lease id generator: {err}");
+        });
+        let id = AtomicU64::new(u64::from_be_bytes(buf));
+        Self { id }
+    }
+
+    /// Generate next id
+    #[inline]
+    pub fn next(&self) -> i64 {
+        let id = self.id.fetch_add(1, Ordering::Relaxed);
+        // to ensure the id is positive
+        if id == 0 {
+            return self.next();
+        }
+        // set the highest bit to 0 as we need only need positive values
+        (id & 0x7fff_ffff_ffff_ffff).cast()
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::collections::HashSet;
+
+    use super::*;
+
+    #[test]
+    fn no_duplicates_in_one() {
+        let mut exist = HashSet::new();
+        let id_gen = LeaseIdGenerator::new();
+
+        for _ in 0..(1 << 20) {
```

Comment:
Does the 20 here has any specific meaning?

Reply:
No it just test it with a rather large number